webapp/channels/src/components/post/post_component.tsx

```diff
@@ -156,29 +157,32 @@ const PostComponent = (props: Props): JSX.Element => {
     }, [a11yActive]);
 
     useEffect(() => {
-        const handleAlt = (e: KeyboardEvent) => {
-            setAlt(e.altKey);
-        };
         let removeEventListener: (type: string, listener: EventListener) => void;
 
-        document.addEventListener('keydown', handleAlt);
-        document.addEventListener('keyup', handleAlt);
         if (postRef.current) {
             postRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
             postRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
             removeEventListener = postRef.current.removeEventListener;
         }
 
         return () => {
-            document.removeEventListener('keydown', handleAlt);
-            document.removeEventListener('keyup', handleAlt);
             if (removeEventListener) {
                 removeEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
                 removeEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
             }
         };
     }, []);
 
+    useEffect(() => {
```

Comment:
@hmhealey I opted for keeping the registering/unregistering the listeners in an effect so that it's clear. The other option is to have them in `handleMouseOver` or `handleMouserLeave` alternatively.

Reply:
Yep, that works as well. It might actually be better since we can track their state easier since there's only one place that they're added/removed

Reply:
I don't think this is actually cleaning up the listeners correctly since we're not removing the event listener when the component is unmounted. I think we might be able to cover both cases if we write it 
```js
useEffect(() => {
    if (hover) {
        // add listeners
    }

    return () => {
         if (hover) {
             // remove listeners
         }
    };
});
```

Reply:
@hmhealey question: shouldn't the first block contain the logic both for adding and removing the listeners and the cleanup function unconditionally include the remove listeners logic?

ie:

```js
useEffect(() => {
        if (hover) {
            document.addEventListener('keydown', handleAlt);
            document.addEventListener('keyup', handleAlt);
        } else {
            document.removeEventListener('keydown', handleAlt);
            document.removeEventListener('keyup', handleAlt);
        }

        return () => {
            document.removeEventListener('keydown', handleAlt);
            document.removeEventListener('keyup', handleAlt);
        }
    }, [hover]);
```